contracts/PolygonTokenBridger.sol

```diff
@@ -88,10 +88,13 @@ contract PolygonTokenBridger is Lockable {
         token.safeTransferFrom(msg.sender, address(this), amount);
 
         // In the wMatic case, this unwraps. For other ERC20s, this is the burn/send action.
-        token.withdraw(amount);
+        token.withdraw(token.balanceOf(address(this)));
 
         // This takes the token that was withdrawn and calls withdraw on the "native" ERC20.
-        if (isWrappedMatic) maticToken.withdraw{ value: amount }(amount);
+        if (isWrappedMatic) {
+            uint256 balance = address(this).balance;
+            maticToken.withdraw{ value: balance }(balance);
+        }
```

Comment:
nit: does this need to be a separate state variable? could it not be:

```suggestion
        if (isWrappedMatic) maticToken.withdraw{ value: address(this).balance }(balance);
```

Reply:
seems reasonable and saves an SSTORE right?

Reply:
I don't think the above works, since balance doesn't exist. Maybe you meant:
```
if (isWrappedMatic) maticToken.withdraw{ value: address(this).balance }(address(this).balance);
```
?

We could do that. I'm not sure it would be cheaper, but that might not matter since this is on polygon and the call is already pretty cheap.

Reply:
Done.

Reply:
ye, sorry. it's hard to structure feedback in these comments with the proposed blocks. what you did is what I had intended.